internal/charmstore/store.go

```diff
@@ -526,9 +526,16 @@ func (s *Store) AddCharm(c charm.Charm, p AddParams) (err error) {
 	if p.URL.URL.Series == "bundle" || p.URL.URL.User == "" || p.URL.URL.Revision == -1 {
 		return errgo.Newf("charm added with invalid id %v", &p.URL.URL)
 	}
-	if p.URL.URL.Series == "" && len(c.Meta().Series) == 0 {
-		return errgo.Newf("charm added without series %v", &p.URL.URL)
+	if len(c.Meta().Series) == 0 {
```

Comment:
I'm not sure we should allow putting a multi-series charm to an id with a series.

So wouldn't this be better as:

     if id.Series == "" && len(c.Meta().Series) == 0 {
          return errgo.WithCausef(nil, params.ErrBadRequest, "charm %v added without any supported series", &p.URL.URL)
     }

Then you don't need the contains logic either.